src/client/components/TablePaginated/Filters/MultiSelect/hooks/useTooltipContent.ts

```diff
@@ -0,0 +1,58 @@
+import { useCallback, useMemo, useState } from 'react'
+
+import { Objects } from 'utils/objects'
+
+import { TablePaginatedFilterType } from 'meta/tablePaginated'
+
+import { useTablePaginatedFilterValue } from 'client/store/ui/tablePaginated/hooks'
+import { TablePaginatedFilter } from 'client/components/TablePaginated/types'
+
+type Props = {
+  fieldName: string
+  options: TablePaginatedFilter<TablePaginatedFilterType.MULTI_SELECT>['options']
+  path: string
+}
+
+type Returned = {
+  hideTooltip: () => void
+  showTooltip: () => void
+  tooltipContent: string | null
+}
+
+export const useTooltipContent = (props: Props): Returned => {
+  const { fieldName, options, path } = props
+  const filterValue = useTablePaginatedFilterValue<Array<string>>(path, fieldName)
+  const [canDisplayTooltip, setCanDisplayTooltip] = useState(true)
+
+  const valueToLabelMap = useMemo(() => {
+    return options.reduce((acc, { value, label }) => {
+      // eslint-disable-next-line no-param-reassign
+      acc[value] = label
+      return acc
+    }, {} as Record<string, string>)
+  }, [options])
+
+  const tooltipContent = useMemo(() => {
```

Comment:
I separated the `valueToLabelMap` because it doesn't depend on the filter selected values. It is calculated based on the select options. If I declare it in this memo, it will be recalculated every time the user changes selection. That's not an issue for this specific filter since it only has a few items, but should we keep it separate in case we have larger selection lists in the future? like countries, although that one will probably be in its own component.